### ISSUE TITLE
Webview does not set a content security policy

### ISSUE THREAD

**Issue by mjbvz** [Azure/vscode-kubernetes-tools#600]
Hi, I'm the developer of VS Code's webview API. I noticed that your extension seems to create a webview that does not set a content security policy. All webviews (even very simple ones) should set a content security policy. This helps limit the potential impact of content injections and is generally a good measure for defense in depth.

We've documented how to add a content security policy to VS Code webviews here. Please add the most restrictive content security policy possible to your webview. I am not aware of any immediate security issues with your extension but having a restrictive content security policy is important to help protect users of your extension.

Also note that in development mode, in VS Code 1.38 you should also see a warning if you create a webview that does not set a content security policy: microsoft/vscode#79248

**Comment by mjbvz** [comment]
Friendly ping. All Microsoft extensions should really have the extra level of protection a strict CSP brings :)

**Comment by itowlson** [comment]
@mjbvz I'm struggling with the documentation and sample - I get `property 'cspSource' does not exist on type 'Webview'` on the img-src fragment, and I don't understand the `script-src 'nonce=${nonce}'` fragment well enough to understand what I should replace it with.

None of my HTML files reference any external resources (though they do need to run JS that is included inline in the HTML, and to talk back to VS Code via the messaging interface).  I'm not sure how to go about expressing this - can you or the VS Code team offer any more specific guidance please?  Thanks!

**Comment by mjbvz** [comment]
Are you using TypeScript? Can you link to your branch?

Content security policies are a general web concept so there's also lots of documentation on how to use them for websites that should help you get started

**Comment by itowlson** [comment]
Yes, I'm using TypeScript.  Here's an example of one of our webviews, picked at random but fairly typical:

https://github.com/Azure/vscode-kubernetes-tools/blob/c34d7d11aaccaad55fcffe126caa14366d053d90/src/components/describe/describeWebview.ts#L71

(That's my master branch; I don't have a branch showing what I'm trying in context.  The fragment I was trying to adapt was from the Cat Coding sample, https://github.com/microsoft/vscode-extension-samples/blob/b5fc559a93bc6326d4f0fef580a19d8c647b1cd9/webview-sample/src/extension.ts#L185.)

Yes, I looked for documentation on CSPs for Web sites (including the link from the VS Code docs), and to be honest I felt overwhelmed.  I don't get how it relates to VS Code, or what threat it mitigates in my context.  I trust the VS Code team when they tell me it's important, but I'm struggling to motivate myself to prioritise a multi-page document that is just the 'introduction'.  Can I just use `'self'` for everything I'm interested in from the list at https://developers.google.com/web/fundamentals/security/csp/#policy_applies_to_a_wide_variety_of_resources and meet your requirements that way?  Thanks!

**Comment by mjbvz** [comment]
Do the VS Code extension samples not work for you? 

CSP are important because—if your html sanitization fails—content injected into a webpage could run scripts or do other bad things. They are a security best practice even if you don't think you will ever be exposed to content injection. 

For your CSP, you should use `none` as the default and then only opt in what you need to (such as running scripts)
